index.js

```diff
@@ -124,21 +124,39 @@ module.exports = function(options) {
     function naughtyHref(href) {
       // Browsers ignore character codes of 32 (space) and below in a surprising
       // number of situations. Start reading here:
-      // https://owasp.org/www-community/xss-filter-evasion-cheatsheet#embedded-tab
+      // https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#Embedded_tab
       // eslint-disable-next-line no-control-regex
       href = href.replace(/[\x00-\x20]+/g, '');
       // Clobber any comments in URLs, which the browser might
       // interpret inside an XML data island, allowing
       // a javascript: URL to be snuck through
-      href = href.replace(/<!--.*?-->/g, '');
+      while (true) {
+        const firstIndex = href.indexOf('<!--');
+        if (firstIndex === -1) {
+          break;
+        }
+        const lastIndex = href.indexOf('-->', firstIndex + 4);
+        if (lastIndex === -1) {
+          break;
+        }
+        href = href.substring(0, firstIndex) + href.substring(lastIndex + 3);
+      }
       // Case insensitive so we don't get faked out by JAVASCRIPT #1
-      var matches = href.match(/^([a-zA-Z]+):/);
+      // Allow more characters after the first so we don't get faked
+      // out by certain schemes browsers accept
+      const matches = href.match(/^([a-zA-Z][a-zA-Z0-9.\-+]*):/);
```

Comment:
There is still a `.` here and `.` matches everything. I don't see any reason to include it?

Reply:
as it stands, this would consider "a:bafflingly:good:time:foobar" to have the protocol "a:bafflingly:good:time" which is definitely not what we want 😄 (* is greedy by default, and we have that `.`, but before we worry about the greediness, why is that `.` there?)